govcd/extension_test.go

```diff
@@ -26,11 +28,132 @@ func (vcd *TestVCD) Test_GetExternalNetwork(check *C) {
+		Description: "Test Create External Network",
+		Xmlns:       types.XMLNamespaceExtension,
+		XmlnsVCloud: types.XMLNamespaceVCloud,
+		Configuration: &types.NetworkConfiguration{
+			Xmlns: types.XMLNamespaceVCloud,
+			IPScopes: &types.IPScopes{
+				IPScope: []*types.IPScope{&types.IPScope{
+					Gateway:   "192.168.201.1",
+					Netmask:   "255.255.255.0",
+					DNS1:      "192.168.202.253",
+					DNS2:      "192.168.202.254",
+					DNSSuffix: "some.net",
+					IPRanges: &types.IPRanges{
+						IPRange: []*types.IPRange{
+							&types.IPRange{
+								StartAddress: "192.168.201.3",
+								EndAddress:   "192.168.201.250",
+							},
+						},
+					},
+				},
+				}},
+			FenceMode: "isolated",
+		},
+		VimPortGroupRefs: &types.VimObjectRefs{
+			VimObjectRef: []*types.VimObjectRef{
+				&types.VimObjectRef{
+					VimServerRef: &types.Reference{
+						HREF: vimServerHref,
+					},
+					MoRef:         portGroups[0].MoRef,
+					VimObjectType: vcd.config.VCD.ExternalNetworkPortGroupType,
+				},
+			},
+		},
+	}
+	task, err := CreateExternalNetwork(vcd.client, externalNetwork)
+	check.Assert(err, IsNil)
+	check.Assert(task.Task, Not(Equals), types.Task{})
+
+	err = task.WaitTaskCompletion()
+	check.Assert(err, IsNil)
+
+	newExternalNetwork, err := GetExternalNetwork(vcd.client, TestCreateExternalNetwork)
+	check.Assert(err, IsNil)
+	check.Assert(newExternalNetwork.ExternalNetwork.Name, Equals, TestCreateExternalNetwork)
+
+	ipScope := newExternalNetwork.ExternalNetwork.Configuration.IPScopes.IPScope
+	check.Assert(ipScope[0].Gateway, Equals, "192.168.201.1")
+	check.Assert(ipScope[0].Netmask, Equals, "255.255.255.0")
+	check.Assert(ipScope[0].DNS1, Equals, "192.168.202.253")
+	check.Assert(ipScope[0].DNS2, Equals, "192.168.202.254")
+	check.Assert(ipScope[0].DNSSuffix, Equals, "some.net")
+
+	check.Assert(len(ipScope[0].IPRanges.IPRange), Equals, 1)
+	ipRange := ipScope[0].IPRanges.IPRange[0]
+	check.Assert(ipRange.StartAddress, Equals, "192.168.201.3")
+	check.Assert(ipRange.EndAddress, Equals, "192.168.201.250")
+
+	check.Assert(newExternalNetwork.ExternalNetwork.Configuration.FenceMode, Equals, "isolated")
+
+	// Needs to be deleted as port group used by other tests
+	// Workaround to refresh until task is fully completed - as task wait isn't enough
+	// Task still exists and creates NETWORK_DELETE error, so we wait until disappears
+	for i := 0; i < 30; i++ {
+		err = newExternalNetwork.Refresh()
+		check.Assert(err, IsNil)
+		if newExternalNetwork.ExternalNetwork.Tasks != nil && len(newExternalNetwork.ExternalNetwork.Tasks.Task) == 0 {
+			break
+		} else {
+			time.Sleep(1 * time.Second)
+		}
+	}
+
+	err = newExternalNetwork.DeleteWait()
+	if err != nil {
+		AddToCleanupList(externalNetwork.Name, "externalNetwork", "", "Test_CreateExternalNetwork")
+	}
+	check.Assert(err, IsNil)
 }
 
 func init() {
-	testingTags["extension"] = "extension_test.go"
+	testingTags["extnetwork"] = "extension_test.go"
```

Comment:
we have extension and external network. Or I don't get idea

Reply:
Idea is that extnetwork tag runs two tests extension and externalnetwork, cause functionality is splitted

Reply:
I second the request. I am partly responsible for naming this file, and it was a short-sighted decision.

`extension_test.go` is only running tests for external network and everything from this file should be moved to `externalnetwork_test.go`.
When we add new functionalities through extensions, it would be better to name the files according to the features we implement rather than the accessor.

Reply:
Moreover, we have two initialization statements for `testingTags["extnetwork"]`. I missed this in my previous review.
When the same feature spans multiple files, we should initialize the tag canary variable only in one of the files.

Reply:
ok. I will move the test, but what about implmenatation `exension.go` - I recommend to move it to system.go and yet, due deprecated method we still have to leave the file for time being.

Reply:
Test moved